api/client/service/ps.go

```diff
@@ -42,13 +42,15 @@ func runPS(dockerCli *client.DockerCli, opts psOptions) error {
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	service, _, err := client.ServiceInspectWithRaw(ctx, opts.serviceID)
-	if err != nil {
-		return err
+	filter := opts.filter.Value()
+	for _, serviceID := range opts.serviceIDs {
+		service, _, err := client.ServiceInspectWithRaw(ctx, serviceID)
```

Comment:
This should require several hits to the client. Use the list API to get the services by prefix (name/id) and _then_ create this call with filters.